### PR TITLE
Refactor to use pygithub

### PR DESCRIPTION
**Pull request by macdiesel**

### CODE REVIEW

**Comment by macdiesel** [comment]
This also includes the update to the end user callable (script called from the pipeline) create_release-candidate.py.

**Comment by macdiesel** [comment]
@edx/pipeline-team PTAL!

File: tubular/tests/test_release.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        self.repo_mock.create_git_ref = create_git_ref_mock
+
+        self.api.create_branch(branch_name, sha)
+
+        create_git_ref_mock.assert_called_with(ref='refs/heads/{}'.format(branch_name), sha=sha)
+        self.repo_mock.get_branch.assert_called_with(branch_name)
+
+    @ddt.data(
+        ('blah-candidate', 'release', 'test', 'test_pr'),
+        ('catnip', 'release', 'My meowsome PR', 'this PR has lots of catnap inside, go crazy!'),
```
Comment (doctoryes) [inline tubular/tests/test_release.py]: Nit: catnap/catnip 😼

File: scripts/create_release_candidate.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    help='The github repository URL',
+    default='edx'
+)
+@click.option(
+    '--repo',
+    help='name of the repository'
+)
+@click.option(
+    '--source_branch',
+    help='the source branch who from which to create the target branch',
```
Comment (doctoryes) [inline scripts/create_release_candidate.py]: Nit: `Source branch whose HEAD is used to create the target branch`?

File: scripts/create_release_candidate.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+@click.command()
+@click.option(
+    '--org',
+    help='The github repository URL',
+    default='edx'
+)
+@click.option(
+    '--repo',
+    help='name of the repository'
```
Comment (doctoryes) [inline scripts/create_release_candidate.py]: Nit: `Name`

File: scripts/create_release_candidate.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    except ValueError:
+        click.BadParameter("Not a valid date: '{0}'.".format(date_str))
+
+EXPECTED_RELEASE_DATE = GitRelease.default_expected_release_date()
+
+
+@click.command()
+@click.option(
+    '--org',
+    help='The github repository URL',
```
Comment (doctoryes) [inline scripts/create_release_candidate.py]: `help` isn't quite correct here. Maybe `Org from the GitHub repository URL of https://github.com/<org>/<repo>`? Or leave out the URL?

File: scripts/create_release_candidate.py (hunk 3 above)
Comment (doctoryes) [reply scripts/create_release_candidate.py]: This `help` should mirror the help above.

File: scripts/create_release_candidate.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    default=None
+)
+@click.option(
+    '--token',
+    envvar='GIT_TOKEN',
+    help='The github access token, see https://help.github.com/articles/creating-an-access-token-for-command-line-use/'
+)
+@click.option(
+    '--output_file',
+    help='File to write the output of the job',
```
Comment (doctoryes) [inline scripts/create_release_candidate.py]: Nit: `File in which to write the script's YAML output`?

**Comment by macdiesel** [comment]
@doctoryes @feanil ready for re-review.

File: scripts/create_release_candidate.py [hunk 6]
```diff
@@ -0,0 +1,226 @@
+"""
```
Comment (feanil) [inline scripts/create_release_candidate.py:1]: This seems slightly different from https://github.com/edx/tubular/pull/83 do we need both?

Comment (macdiesel) [reply scripts/create_release_candidate.py:1]: One will create a branch and PR, the other simply creates the branch. The requirements have changed a bit while this is in flight. I would like to get this in today to deal with the branch cutting things the we need for the release-candidate today then make a followup ticket to come back and modify this with the new requirements we may have once we get this a bit more sorted out.

Comment (feanil) [reply scripts/create_release_candidate.py:1]: Sounds good to me.

**Review by feanil — APPROVED** [review]

**Pull request merged**